tools/cell_census_builder/scripts/release/census_summary.py

```diff
@@ -28,3 +30,64 @@
     stats_df = pd.DataFrame(stats, columns=["organism", "attribute", "unique count"])
     display_stats_df = pd.pivot(stats_df, index=["organism"], columns=["attribute"], values=["unique count"])
     print(display_stats_df)
+
+    # Calculate the diff with respect to the previous version (if specified)
+
+    if previous_census_version is not None:
+        previous_census = cell_census.open_soma(census_version=previous_census_version)
+
+        prev_datasets = previous_census[CENSUS_INFO_NAME]["datasets"].read().concat().to_pandas()
+        curr_datasets = census[CENSUS_INFO_NAME]["datasets"].read().concat().to_pandas()
+
+        # Datasets removed, added
+        curr_datasets_ids = set(curr_datasets["dataset_id"])
+        prev_dataset_ids = set(prev_datasets["dataset_id"])
+
+        added_datasets = curr_datasets_ids - prev_dataset_ids
+        removed_datasets = prev_dataset_ids - curr_datasets_ids
+        if added_datasets:
+            print("Datasets that were added")
+            for d in added_datasets:
+                print(d)
+
+        if removed_datasets:
+            print("Datasets that were removed")
+            for d in removed_datasets:
+                print(d)
+
+        # Datasets in both versions but that have differing cell counts
+        joined = prev_datasets.join(
+            curr_datasets.set_index("dataset_id"), on="dataset_id", lsuffix="_prev", rsuffix="_curr"
+        )
+        datasets_with_different_cell_counts = joined.loc[
+            joined["dataset_total_cell_count_prev"] != joined["dataset_total_cell_count_curr"]
+        ][["dataset_id", "dataset_total_cell_count_prev", "dataset_total_cell_count_curr"]]
+
+        if not datasets_with_different_cell_counts.empty:
+            print("Datasets that have a different cell count")
+            print(datasets_with_different_cell_counts)
+
+        # Total cell count deltas by experiment (mouse, human)
+        for organism in ["homo_sapiens", "mus_musculus"]:
```

Comment:
there is a slightly more future-proof way to do this, as anything inside `census['census_data']` is an Experiment.  So I suggest:

```
for organism_name in census[CENSUS_DATA]:
    curr_count = census[CENSUS_DATA][organism_name].obs.count
    ...
```

Or even better:
```
for organism_name, experiment in census[CENSUS_DATA].items():
    curr_count = experiment.obs.count
    ...
```

Also of note:  pre-defined fields like `obs` are attributes, so you can use `experiment.obs`, rather than the more verbose `experiment["obs"]`